src/topology/algebra/semigroup.lean

```diff
@@ -25,50 +25,45 @@ lemma exists_idempotent_of_compact_t2_of_continuous_mul_left {M} [nonempty M] [s
 begin
 /- We apply Zorn's lemma to the poset of nonempty closed subsemigroups of `M`. It will turn out that
 any minimal element is `{m}` for an idempotent `m : M`. -/
-  let S : set (set M) := { N : set M | is_closed N ∧ N.nonempty ∧ ∀ m m' ∈ N, m * m' ∈ N },
+  let S := {N : set M | is_closed N ∧ N.nonempty ∧ ∀ m m' ∈ N, m * m' ∈ N},
```

Comment:
I don't think that removing explicit type annotations to save a few symbols is a good idea. They make the proof more readable.

Reply:
I can still remove the type annotation on `N` itself, though? After all, it's made redundant by the type annotation on `S`.